samples/snippets/core/testing/unit-testing-best-practices/csharp/unit-testing-best-practices/StringCalculator.cs

```diff
@@ -10,7 +10,7 @@ public int Add(string numbers)
         {
             if (numbers == null)
             {
-                throw new ArgumentException(nameof(numbers));
+                throw new ArgumentException(null, nameof(numbers));
```

Comment:
```suggestion
                throw new ArgumentNullException(nameof(numbers));
```

What about something like this instead? This should really be an `ArgumentNullException`.

Reply:
Ah, that is clearly better. Did not know about the ArgumentNullException.